liblouis/lou_translateString.c

```diff
@@ -378,9 +378,15 @@ swapReplace(int start, int end, const TranslationTableHeader *table, InString in
 		int rep;
 		int test;
 		int k;
-		for (test = 0; test < swapRule->charslen; test++)
-			if (input.chars[p] == swapRule->charsdots[test]) break;
-		if (test == swapRule->charslen) continue;
+		if (swapRule->opcode == CTO_SwapDd) {
+			for (test = 0; test * 2 + 1 < swapRule->charslen; test++)
```

Comment:
Where do these magic numbers come from?

Reply:
A sequence of dot patterns is encoded as the length of the first dot pattern (single widechar) followed by the contents of the first dot pattern (one widechar per cell) followed by the length of the second dot pattern, etc. See `compileSwapDots`. Because the third operand of a swapdd rule can only contain single-cell dot patterns, it becomes `[1, <1st dots>, 1, <2nd dots>, ...]`.